lib/lambda_ethereum_consensus/validator/duties.ex

```diff
@@ -73,21 +84,26 @@ defmodule LambdaEthereumConsensus.Validator.Duties do
         new_proposers = compute_proposers_for_epoch(beacon, epoch, validators)
         new_attesters = compute_attesters_for_epoch(beacon, epoch, validators)
 
-        new_sync_committees =
+        {new_sync_committees, new_misc_data} =
           case sync_committee_compute_check(epoch, {last_epoch, Map.get(duties_map, last_epoch)}) do
-            {:already_computed, sync_committees} ->
-              sync_committees
+            {:already_computed, sync_committee_duties} ->
+              sync_committee_duties
+              |> recompute_sync_committee_duties(beacon, epoch, validators)
+              |> then(&{&1, misc_data(duties_map, last_epoch)})
 
             {:not_computed, period} ->
               Logger.debug("[Duties] Computing sync committees for period: #{period}.")
 
-              compute_current_sync_committees(beacon, validators)
+              beacon
+              |> compute_sync_committee_duties(epoch, validators)
+              |> then(&{&1, compute_misc_data(beacon, epoch)})
```

Comment:
If this is the heavy one that calculates all participants for an epoch we probably shouldn't do it inside this for.

Reply:
This is indeed inside the epoch for `for {epoch, slot} <- epochs_and_start_slots, reduce: duties_map do`, which normally is just one epoch and two at the beginning when we don't have any lookahead. But it's also inside a case:

```elixir
   case sync_committee_compute_check(epoch, {last_epoch, Map.get(duties_map, last_epoch)}) do
      {:already_computed, sync_committee_duties} ->
        sync_committee_duties
        |> recompute_sync_committee_duties(beacon, epoch, validators)
        |> then(&{&1, misc_data(duties_map, last_epoch)})

      {:not_computed, period} ->
        Logger.debug("[Duties] Computing sync committees for period: #{period}.")

        beacon
        |> compute_sync_committee_duties(epoch, validators)
        |> then(&{&1, compute_misc_data(beacon, epoch)})
    end
```

sync_committee_compute_check/2, makes sure we didn't change epoch since our previous epoch duty calculation, it just hit the  `{:not_computed, period} ->` when we get into a new period. Right now is the only thing that's calculated, but I'll rename the function to let us know its just calculating the participants.

Reply:
renamed to `compute_sync_subcommittee_participants` as discussed